ledger/applications.go

```diff
@@ -76,7 +78,7 @@ func (ae *appTealEvaluator) InitLedger(balances transactions.Balances, acctWhite
 	return nil
 }
 
-func newAppLedger(balances transactions.Balances, acctWhitelist []basics.Address, appGlobalWhitelist []basics.AppIndex, appIdx basics.AppIndex) (al *appLedger, err error) {
+func newAppLedger(balances transactions.Balances, acctWhitelist []basics.Address, appGlobalWhitelist []basics.AppIndex, appIdx basics.AppIndex, latestTimestamp int64) (al *appLedger, err error) {
```

Comment:
Currently we're getting `Round` through the `transactions.Balances` interface, which is tied to a particular round. I think we should only access ledger information through that interface (like we do for round), or only pass them as arguments, but a mixture of both looks messy to me.

Reply:
It seems weird to pass the latest timestamp through Balances. It seems to make more sense to pass the round separately from Balances. Though I think you could argue that the Round is special in that Balances also needs it (for checking whether a transaction is alive)?

Reply:
My thinking was that `transactions.Balances` is currently the only way that the `appLedger` methods access the state of the world outside of this transaction group -- since `latestTimestamp` and `round` are always uniquely determined by a particular instance of `Balances` it felt redundant/potentially error prone to pass them separately.

Reply:
Yeah. I think originally `transactions.Balances` was a way to pass state to `Transaction.Apply`, and the round was needed there to check for transaction liveness. The latest timestamp is not needed in transaction application, but it is needed in TEAL evaluation. I think it would also be correct to pass it through `Balances`, but from the perspective of `Transaction.Apply`, it would be a little strange.

Reply:
I put both round and latest timestamp into a struct, as discussed.